src/lib/communication/DY.Lib.Communication.RequestResponse.Invariants.fst

```diff
@@ -192,21 +227,20 @@ let request_response_event_preconditions #cinvs = {
 
 val event_predicate_communication_layer_reqres_and_tag:
   {|cinvs:crypto_invariants|} ->
-  #a:Type -> {| parseable_serializeable bytes a |} ->
+  #a:Type0 -> {| comm_layer_reqres_config a |} ->
   comm_reqres_higher_layer_event_preds a ->
   list (string & compiled_event_predicate)
 let event_predicate_communication_layer_reqres_and_tag #cinvs #a higher_layer_resreq_preds =
   [
-    event_predicate_communication_layer_and_tag request_response_event_preconditions;
+    event_predicate_communication_layer_core_and_tag (comm_core_higher_layer_event_preds_reqres a);
     mk_event_tag_and_pred (event_predicate_communication_layer_reqres higher_layer_resreq_preds)
   ]
 
 val has_communication_layer_reqres_event_predicates:
   {|protocol_invariants|} ->
-  #a:Type -> {| parseable_serializeable bytes a |} ->
-  comm_higher_layer_event_preds com_message_t ->
+  #a:Type0 -> {| comm_layer_reqres_config a |} ->
   comm_reqres_higher_layer_event_preds a ->
   prop
-let has_communication_layer_reqres_event_predicates #invs #a higher_layer_preds higher_layer_resreq_preds =
-  has_event_pred (event_predicate_communication_layer #invs.crypto_invs higher_layer_preds) /\
-  has_event_pred (event_predicate_communication_layer_reqres #invs.crypto_invs higher_layer_resreq_preds)
+let has_communication_layer_reqres_event_predicates #invs #a #config higher_layer_resreq_preds =
+  has_event_pred (event_predicate_communication_layer_core (comm_core_higher_layer_event_preds_reqres a)) /\
+  has_event_pred (event_predicate_communication_layer_reqres higher_layer_resreq_preds)
```

Comment:
Note: It might make sense to have a general `has_communication_layer_reqres_predicates` that includes both the event and state predicates (and crypto, if applicable). Same thing for core if it doesn't already exist.